src/Diagnostics.DataProviders/DataProviders/KustoDataProvider.cs

```diff
@@ -4,26 +4,120 @@
 using System.Collections.Concurrent;
 using System.Collections.Generic;
 using System.Data;
+using System.IO;
+using System.IO.Compression;
 using System.Linq;
+using System.Runtime.CompilerServices;
 using System.Text;
 using System.Threading.Tasks;
+using System.Web;
 
 namespace Diagnostics.DataProviders
 {
+    public class KustoQuery
+    {
+        public string Text;        
+        public string Url;
+    }
     public class KustoDataProvider: DiagnosticDataProvider, IDiagnosticDataProvider
     {
         private KustoDataProviderConfiguration _configuration;
         private IKustoClient _kustoClient;
 
-        public KustoDataProvider(OperationDataCache cache, KustoDataProviderConfiguration configuration): base(cache)
+        public KustoDataProvider(OperationDataCache cache, KustoDataProviderConfiguration configuration) : base(cache)
         {
             _configuration = configuration;
             _kustoClient = KustoClientFactory.GetKustoClient(configuration);
+            Metadata = new DataProviderMetadata
```

Comment:
I am fine with this approach as long as we are not using KustoDataProvider (or any data provider) as a singleton service.
This is a good approach to not break the existing contracts.

Reply:
I realized that DataProviders were not singleton so I thought this makes sense. Please let me know if anything else required from my end for this PR. I did realize that one method (ParseRegionFromStamp) is duplicated and I am open to discussion regarding the same